packages/backend/src/api/jobScraping/scraping.ts

```diff
@@ -0,0 +1,353 @@
+    jobLinks.push(href);
+  }
+
+  const jobs: any[] = [];
+
+  for (const link of jobLinks) {
+    await page.goto(link);
+    const html = await page.content();
+    const $ = cheerio.load(html);
+
+    const title = $('.cw-jobheader-info > h1:nth-child(2)').text().trim();
+    const description = $('#spandesc > ul:nth-child(2) > li , #spanreq > ul:nth-child(2) > li').text().trim();
+
+    const { languages, frameworks } = filterLanguagesAndFrameworks(description);
+
+    const applicationLink = await shortenLink(link);
+    jobs.push({ title, languages, frameworks, applicationLink });
+  }
+  return jobs;
+};
+
+export const smartRecruitersScrape = async () => {
+  const browser = await puppeteer.launch();
+  const page = await browser.newPage();
+
+  page.goto(smartRecruiters);
+
+  await page.waitForSelector('.jobs-item');
+
+  const jobLinks: any[] = [];
+
+  const jobListings = await page.$$('.jobs-item');
+  for (const link of jobListings) {
+    const href = await link.$eval('a', (el) => el.href);
+    jobLinks.push(href);
+  }
+
+  const jobs: any[] = [];
+
+  for (const link of jobLinks) {
+    await page.goto(jobLinks[1]);
+    const html = await page.content();
+    const $ = cheerio.load(html);
+
+    const title = $('.job-title').text().trim();
+    const description = $(
+      '#st-companyDescription > div:nth-child(2) > p , #st-jobDescription > div:nth-child(2) > p, #st-jobDescription > div:nth-child(2) > ul:nth-child(3) > li , #st-additionalInformation > div:nth-child(2) > ul:nth-child(2) > li'
+    )
+      .text()
+      .trim();
+    const { languages, frameworks } = filterLanguagesAndFrameworks(description);
+
+    const applicationLink = await shortenLink(link);
+
+    jobs.push({ title, languages, frameworks, applicationLink });
+  }
+
+  return jobs;
+};
+
+export const baytScrape = async () => {
+  const browser = await puppeteer.launch();
+  const page = await browser.newPage();
+
+  page.goto(bayt);
+
+  await page.waitForSelector('#results_inner_card');
+
+  const jobLinks: any[] = [];
+
+  const jobListings = await page.$$('.has-pointer-d');
+  for (const link of jobListings) {
+    const href = await link.$eval('a', (el) => el.href);
+    jobLinks.push(href);
+  }
+
+  const jobs: any[] = [];
+
+  for (const link of jobLinks) {
+    await page.goto(link);
+    const html = await page.content();
+    const $ = cheerio.load(html);
+
+    const title = $('#job_title').text().trim();
+    const description = $('div.t-break').text().trim();
+
+    const { languages, frameworks } = filterLanguagesAndFrameworks(description);
+
+    const applicationLink = await shortenLink(link);
+
+    jobs.push({ title, languages, frameworks, applicationLink });
+  }
+  return jobs;
+};
+
+//TODO:fix filtering the locations to get the remote jobs only
+export const remoteSourceScrape = async () => {
+  const browser = await puppeteer.launch();
+  const page = await browser.newPage();
+
+  page.goto(remoteSource);
```

Comment:
those lines are redundant, would be nice to have the implementation located in one helper method that's used across all the concrete classes